File: Sample codes/Online Card Present Verification/Python/online_card_present_verification.py
import hashlib import requests import  json
import collections

url = 'https://api.shuftipro.com/
CLIENT_ID = 'Your client id provided by Shuftipro'
SECRET_KEY = 'YOUR_SECRET_KEY'

post_data = {
"method"              : "id_card OR passport OR driving_license",
"client_id"           : CLIENT_ID, "card_first_6_digits" : "123456", "card_last_4_digits"  : "7890",
"reference"           : "Your unique request reference", "country"             : "Pakistan",
"phone_number"        : "+440000000000",
"callback_url"        : "A valid callback url e.g https://www.yourdomain.com", "redirect_url"        : "A valid callback url e.g https://www.yourdomain.com",
}


post_data = collections.OrderedDict(sorted(post_data.items())) #sort the dictionary raw_data = "".join(post_data.values()) + SECRET_KEY #get values from dictionary and append secret key

hash_object = hashlib.sha256(raw_data) #calculating sha 256 hash signature = hash_object.hexdigest()

post_data['signature'] = signature #append signature to data dictionary response = requests.post(url, post_data).json() #send POST request to API

if response['status_code'] == "SP2":
print response['message']   #now you can redirect your customer to this url

File: Sample codes/Online Card Present Verification/PHP/online_card_present_verification.php
<?php
$url="https://api.shuftipro.com/";

$verification_services = array(
	"document_type"        => "passport",
	"document_id_no"       => "123-ABC-001",
	"document_expiry_date" => "2025-01-01",
	"address"              => "your address",
	"card_first_6_digits"  => "123456",
	"card_last_4_digits"   => "7890",
	"background_checks"    => "0"
);
//JSON encode the services array
$verification_services = json_encode($verification_services);

$post_data = array(
	"client_id"            => $client_id,
	"reference"            => "Your unique request reference",
	"email"                => "customer email",
	"phone_number"         => "+440000000000",
	"country"              => "Pakistan",
	"lang"				   => "2 digits code of supported languages for intarface language"
	"callback_url"         => "A valid callback url e.g https://www.yourdomain.com", 
	"redirect_url"         => "A valid callback url e.g https://www.yourdomain.com",
	"verification_services" => $verification_services,    

);

ksort($post_data);//Sort the all request parameter.
$raw_data = implode("", $post_data) . "YOUR_SECRET_KEY"; //Replace with your secret key provided by the Shuftipro;

$signature              = hash("sha256", $raw_data);
$post_data["signature"] = $signature;

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url); curl_setopt($ch, CURLOPT_POST, 1); curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data); curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$response = curl_exec($ch);
curl_close($ch);
?>